marconi-chain-index/src/Marconi/ChainIndex/Experimental/Run.hs

```diff
@@ -66,9 +68,32 @@ run appName = withGracefulTermination_ $ do
       retryConfig = Cli.optionsRetryConfig $ Cli.commonOptions o
       preferredStartingPoint = Cli.optionsChainPoint $ Cli.commonOptions o
 
+  {-
+      Logging in 'Cardano.BM' works by putting items on a switchboard queue and processing them
+      asynchronously.
+
+      In order to ensure that we always get errors written to the console, we need to wait for the
+      queue to complete before exiting. We can use the 'shutdown' function to guarantee this.
+
+      'shutdown' puts a kill-pill on the switchboard's queue, then waits for the switchboard's
+      dispatcher to exit, which requires the queue to be drained, ensuring all messages are
+      processed.
+  -}
+  let informativeFailure :: Text -> IO a
```

Comment:
informativeFailure -> logInformativeFailure? Or maybe just logFullError?

Reply:
Hmmm, it was because I'm also calling `exitFailure` here. Maybe I should just remove the `exitFailure` call and use your naming suggestion.